### PR TITLE
Toctree: don't add url fragment for h1

### LINKED ISSUE #137: Don't add url fragment "#" in toc tree of level 1
Hello,

this is related to https://github.com/weaverryan/docs-builder/issues/42

I'm wondering if you'd be ok with this fix? This seems logic not to add the fragment when toc are linking to page and not an anchor inside the page.

It would be an easy fix:
```diff
// ./lib/HTML/Renderers/TocNodeRenderer.php

-    private function generateTarget(?string $url, $title): array
+    private function generateTarget(?string $url, $title, int $level): array
    {
-        $anchor = $this->generateAnchorFromTitle($title);
-
-        $target = $url . '#' . $anchor;
+        $target = $url;
+        if ($level > 1) {
+            $anchor = $this->generateAnchorFromTitle($title);
+            $target .= '#' . $anchor;
+        }
```

thanks.

### PR DESCRIPTION
**Pull request by nikophil**
fixes #137

### CODE REVIEW

File: tests/Builder/BuilderTest.php [hunk 1]
```diff
@@ -158,36 +158,46 @@ public function testToctreeGlob(): void
     {
         $contents = $this->getFileContents($this->targetFile('toc-glob.html'));
 
-        self::assertStringContainsString('magic-link.html#another-page', $contents);
-        self::assertStringContainsString('introduction.html#introduction-page', $contents);
-        self::assertStringContainsString('subdirective.html', $contents);
+        // links to first <h1> to other pages does not contains url fragment
```
Comment (greg0ire) [inline tests/Builder/BuilderTest.php]:
```suggestion
        // links to first <h1> tag of other pages must not contain a url fragment
```

File: tests/Builder/BuilderTest.php [hunk 2]
```diff
@@ -158,36 +158,46 @@ public function testToctreeGlob(): void
...
+        self::assertStringContainsString('magic-link.html"', $contents);
+        self::assertStringContainsString('introduction.html"', $contents);
+        self::assertStringContainsString('subdirective.html"', $contents);
+        self::assertStringContainsString('subdir/file.html"', $contents);
+
+        // links to other <h1> should contain url fragment
```
Comment (greg0ire) [inline tests/Builder/BuilderTest.php]:
```suggestion
        // links to other <h1> tags should contain a url fragment
```

File: tests/Builder/BuilderTest.php [hunk 3]
```diff
@@ -158,36 +158,46 @@ public function testToctreeGlob(): void
...
+        self::assertStringContainsString('index.html#another-h1', $contents);
+
+        // links to other headings contains url fragment
```
Comment (greg0ire) [inline tests/Builder/BuilderTest.php]:
```suggestion
        // links to other headings contain a url fragment
```

File: tests/Builder/BuilderTest.php [hunk 4]
```diff
@@ -158,36 +158,46 @@ public function testToctreeGlob(): void
...
         self::assertStringContainsString('subdir/test.html#subdirectory', $contents);
-        self::assertStringContainsString('subdir/file.html#heading-1', $contents);
+        self::assertStringContainsString('subdir/file.html#heading-2', $contents);
+
+        // link to <h1> inside the same page contains url fragment
```
Comment (greg0ire) [inline tests/Builder/BuilderTest.php]:
```suggestion
        // link to <h1> inside the same page contains a url fragment
```

**Review by greg0ire — APPROVED** [review]

**Comment by greg0ire** [comment]
@wouterj please review

**Review by wouterj — APPROVED** [review]
As far as I know, @nikophil understands this package much better than I do. And the tests make sense, so I'm definitely :+1:

**Comment by greg0ire** [comment]
@wouterj yes, but I think you understand the package much better than I do, hence why I ask you to review it. I hope you don't mind :stuck_out_tongue:

**Comment by greg0ire** [comment]
Thanks @nikophil !

**Pull request merged**